tests/migration_test.go

```diff
@@ -2701,65 +2701,76 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
 
-				var originalNodeLabels map[string]string
-				var originalNodeAnnotations map[string]string
+				var originalNodeLabels []map[string]string
+				var originalNodeAnnotations []map[string]string
 				var vmi *v1.VirtualMachineInstance
-				var node *k8sv1.Node
+				var backupNodes []k8sv1.Node
 
 				BeforeEach(func() {
+					backupNodes = libnode.GetAllSchedulableNodes(virtClient).Items
+					if len(backupNodes) == 1 || len(backupNodes) > 10 {
+						Skip("This test can't run with single node and it's too slow to run with more than 10 nodes")
+					}
+					originalNodeLabels = make([]map[string]string, len(backupNodes))
+					originalNodeAnnotations = make([]map[string]string, len(backupNodes))
 					By("Creating a VMI with default CPU mode")
 					vmi = alpineVMIWithEvictionStrategy()
 					vmi.Spec.Domain.CPU = &v1.CPU{Model: v1.CPUModeHostModel}
 
 					By("Starting the VirtualMachineInstance")
 					vmi = runVMIAndExpectLaunch(vmi, 240)
 
-					By("Saving the original node's state")
-					node, err = virtClient.CoreV1().Nodes().Get(context.Background(), vmi.Status.NodeName, metav1.GetOptions{})
-					Expect(err).ToNot(HaveOccurred())
-
-					originalNodeLabels = copyMap(node.Labels)
-					originalNodeAnnotations = copyMap(node.Annotations)
-
-					node = disableNodeLabeller(node, virtClient)
+					for indx, node := range backupNodes {
+						originalNodeLabels[indx] = copyMap(node.Labels)
+						originalNodeAnnotations[indx] = copyMap(node.Annotations)
+						disableNodeLabeller(&node, virtClient)
+					}
 				})
 
 				AfterEach(func() {
 					By("Restore node to its original state")
-					node.Labels = originalNodeLabels
-					node.Annotations = originalNodeAnnotations
-					node, err = virtClient.CoreV1().Nodes().Update(context.Background(), node, metav1.UpdateOptions{})
-					Expect(err).ShouldNot(HaveOccurred())
-
-					Eventually(func() map[string]string {
-						node, err = virtClient.CoreV1().Nodes().Get(context.Background(), node.Name, metav1.GetOptions{})
+					for indx, node := range backupNodes {
+						currNode, err := virtClient.CoreV1().Nodes().Get(context.Background(), node.Name, metav1.GetOptions{})
 						Expect(err).ShouldNot(HaveOccurred())
+						currNode.Labels = originalNodeLabels[indx]
+						currNode.Annotations = originalNodeAnnotations[indx]
+						_, err = virtClient.CoreV1().Nodes().Update(context.Background(), currNode, metav1.UpdateOptions{})
+						Expect(err).ShouldNot(HaveOccurred())
+					}
+					for indx, node := range backupNodes {
+						var currNode *k8sv1.Node
+						Eventually(func() map[string]string {
+							currNode, err = virtClient.CoreV1().Nodes().Get(context.Background(), node.Name, metav1.GetOptions{})
+							Expect(err).ShouldNot(HaveOccurred())
+							return currNode.Labels
+						}, 10*time.Second, 1*time.Second).Should(Equal(originalNodeLabels[indx]), "Node should have fake host model")
+						Expect(currNode.Annotations).To(Equal(originalNodeAnnotations[indx]))
+					}
```

Comment:
the reason i added another `for` loop is for Performance considerations.
in the first loop we update the nodes and than we verify that all the nodes have the original labels and Annotations
Instead of updating a node and wait for it to propagate to the `etcd` for every node.

Reply:
Makes sense, thanks!